nRF RPC multi-instance transport support (#728)
nrfconnect/sdk-nrfxlib: nrf_rpc/doc/architecture.rst

---------

The main role of a transport is to transfer packets between two sides.
You can implement your own transport for any nRF RPC group.
The transport layer defines a destination remote CPU.

The header file describing the nRF RPC transport API is :file:`include/nrf_rpc_tr.h`.

> In the next step, you need to assign your transport interface to the nRF RPC group in the following way:

Reviewer: ```suggestion
Assign your transport interface to the nRF RPC group as follows:
```

Author: Done
